Make the `make build` depends on `npm_licenses` task

showuon opened the pull request:
Fix #7812 . After investigation, the reason why we ship the NPM licenses with the docker image, but not in the tarball built by promu, is because we forgot to add the `npm_licenses` dependency to the `build` task. 

In `promu crossbuild`, we will run `make build` to build the project. But the `npm_licenses` was not the build dependent to the `build` task, so it won't include in the release tarball. On the other hand, the `npm_licenses` is the dependency of the `docker` task, and that's why it'll include in the docker image. 
```Makefile
.PHONY: docker
docker: npm_licenses common-docker
```

I added the `npm_licenses` task as the `build` task dependency, and add some comments in the `.promu.yml` to remind other developers.

SuperQ commented on .promu.yml:
```diff
@@ -18,6 +18,8 @@ build:
         -X github.com/prometheus/common/version.BuildDate={{date "20060102-15:04:05"}}
 tarball:
+    # Whenever there are new files to include in the tarball,
+    # remember to make sure the new files will be generated after `make build`
```
Minor but, end with `.`

showuon replied:
@SuperQ , thanks for reminding. My bad habit! Just updated. Thanks.

roidelapluie commented:
Thanks for your PR. For legal reasons, we require that all commits are signed with a [DCO](https://developercertificate.org/) before we can merge them. See [this blog post](https://www.cncf.io/blog/2016/11/08/cloud-native-software-can-trust/) for considerations around this.

This means that the last line of your commit message should read like:

`Signed-Off-By: Your Name <your@email.address>`

If you are using GitHub through the web interface, it's quickest to close this PR and open a new one with the appropriate line.

If you are using Git on the command line, it is probably quickest to amend and force push. You can do that with
````
git commit --amend --signoff
git push -f $remote $remote_branch_for_pr
````

As always, be careful when force-pushing.

showuon commented:
@roidelapluie , thanks for your reminding! I've updated.
@SuperQ , I found after I added the dependency to the `build` task, the user not using Linux OS cannot run `make build` successfully anymore. It's because we use `tar --transform`, which is not work in macOS (BSD-tar). Here's the [reference](https://github.com/Psy-Fer/fast5_fetcher/issues/4). And not to mention that in Windows, there's even no `rm` command available. So, I added the OS check before running `npm_licenses` task ([reference](https://stackoverflow.com/a/12099167)). Ignoring the `npm_licenses` task can still make the build and everything works well as before.
Thanks.

roidelapluie commented:
Which windows user would have make but not RM?

showuon commented:
> Which windows user would have make but not RM?

Haha, yeah, it could be. I'm not sure if this is a good solution for it, or we can just keep it as is and let users handle it by themselves. See if you have any suggestions. Thanks.

showuon commented:
@SuperQ , need your comments about if we need to let the non-Linux users skipping the `npm_licenses`? Since we make `build` depends on the `npm_licenses`  now, it might make non-Linux OS users build failed. I'm thinking they should skip the `npm_licenses` task since it's only for tarball release use. How do you think? 
Thanks.

showuon commented:
Hi @SuperQ , we're still waiting for your comments about this issue. Thanks.

showuon commented:
Sorry, @SuperQ  , we're still waiting for your comments about this issue. Thanks.

showuon commented:
hi @roidelapluie  , looks like @SuperQ  is busy on other things.  I think this PR is only 1 decision left: should we add OS checks during running `make build` (for `npm_licenses` task)? Maybe you can help provide suggestion for this PR to move on. Thanks.

Options:
1. **Add OS check to only allow Linux OS to run npm_licenses task**: This is my preference and current PR status, since the `npm_licenses` only needed in our release build, not for development. And our release build machine is a Linux OS(Linux-amd64).

2. **Don't do OS check**: Developers should have the ability to troubleshooting the issue by themselves if needed.

showuon commented:
@roidelapluie @SuperQ , need your comment on this PR. Thanks.

showuon commented:
@roidelapluie @SuperQ, any comments?

Nexucis commented:
Reviving a bit this PR. Sorry for the long time to reply.

I think the simpler, the better is it. So I would say the same rule should apply to anybody regardless the OS.
I agree with @roidelapluie, if you are able to use `make` in windows, you are able to use `rm` as well.

And I think if `npm_licenses` is a dependency of `build` then we could remove it from `docker` and `tarball`

Nexucis commented:
if you can rebase the PR and remove the check of the OS @showuon we should be able to merge this PR.

showuon commented:
@Nexucis , PR updated. Thanks.

Nexucis reviewed (approved):
thanks @showuon

Nexucis commented:
triggering CircleCI jobs
